src/components/container/SeverityCalendar/SeverityCalendar.tsx

```diff
@@ -0,0 +1,158 @@
+import queryAllSurveyAnswers from "../../../helpers/query-all-survey-answers";
+import Calendar from "../../presentational/Calendar/Calendar";
+import { format, parseISO, startOfDay, startOfMonth } from "date-fns";
+import { CalendarDay, CalendarDayStateConfiguration, DateRangeContext, LoadingIndicator } from "../../presentational";
+import { previewSeverityData } from "./SeverityCalendar.previewdata";
+import { useInitializeView } from "../../../helpers/Initialization";
+export type SeverityCalendarPreviewState = "Default" | "NoData";
+
+export interface SeverityCalendarProps {
+    surveyName: string,
+    dateRecordedResultIdentifier: string,
+    severityResultIdentifier: string,
+    intervalStart?: Date,
+    previewState?: SeverityCalendarPreviewState,
+}
+
+interface SeverityLogEntry {
+    dateObserved: Date,
+    severity: string,
+    dateEntered: string
+}
+
+export default function (props: SeverityCalendarProps) {
+    const surveyAnswerQuery: SurveyAnswersQuery = {
+        surveyName: props.surveyName,
+        resultIdentifier: [props.dateRecordedResultIdentifier, props.severityResultIdentifier]
+    }
+
+    const [data, setData] = useState<Map<string, SeverityLogEntry> | undefined>();
+    const dateRangeContext = useContext(DateRangeContext);
+    let intervalStart = dateRangeContext?.intervalStart ?? props.intervalStart ?? startOfMonth(new Date());
+
+    async function initialize() {
+        if (props.previewState) {
+            transformToCalendarData(props.previewState === "Default" ? previewSeverityData : []);
+        } else {
+            queryAllSurveyAnswers(surveyAnswerQuery).then((results: SurveyAnswer[]) => {
+                transformToCalendarData(results);
+            });
+        }
+    }
+
+    function transformToCalendarData(results: SurveyAnswer[]) {
+
+        var resultByDateMap = new Map<string, SeverityLogEntry>();
+        var groupedByResult = groupAnswersByResult(results);
+
+        groupedByResult.forEach((grouping) => {
+            var severity = "";
+            if (grouping.severityAnswer) {
+                switch (grouping.severityAnswer.answers[0]) {
+                    case "mild":
+                    case "severity1":
+                        severity = "mild";
+                        break;
+                    case "moderate":
+                    case "severity2":
+                        severity = "moderate";
+                        break;
+                    case "severe":
+                    case "severity3":
+                        severity = "severe";
+                        break;
+                }
+
+                if (severity) {
+                    let dateOfSeverity = parseISO(grouping.dateObservedAnswer ? grouping.dateObservedAnswer.answers[0] : grouping.severityAnswer.date);
+                    let key = format(startOfDay(dateOfSeverity), 'MM/dd/yyyy');
+                    let exists = resultByDateMap.get(key);
+                    if (exists) {
```

Comment:
I think you might be able to simplify this conditional update as: 

```
if (!exists || dateOfSeverity > exists.dateObserved || (dateOfSeverity === exists.dateObserved && grouping.severityAnswer.date > exists.dateEntered)){
	resultByDateMap.set(key, { dateObserved: dateOfSeverity, severity: severity, dateEntered: grouping.severityAnswer.date });
}
```

Reply:
Thinking about this a tad more.  Is the `dateOfSeverity > exists.dateObserved` condition even necessary?  If the `key` and `dateObserved` are always set to the same value, which is the `dateOfSeverity`, then I'm not sure how the `dateOfSeverity` could ever be greater than the `exists.dateObserved`.

Am I missing something in the logic here?

Reply:
I *think* `dateOfSeverity` and `exists.dateObserved` will always be equal.  So maybe this can simplify even further to: 

```
if (!exists || grouping.severityAnswer.date > exists.dateEntered){
	resultByDateMap.set(key, { dateObserved: dateOfSeverity, severity: severity, dateEntered: grouping.severityAnswer.date });
}
```

Reply:
The severity entry can come in with and without a dedicated date observed result.
You are right that with the key being the dateObserved that testing if the date is greater than has no value. Good catch. 
If the user enters multiple entries for the same Observed date then the most recent needs to be used.

Reply:
I think the `dateOfSeverity === exists.dateObserved` condition will always be true as well, given how the data is set.  The `key` and `dateObserved` are always set to the same thing, so any time the `key` from a given `dateOfSeverity` is used to look up an existing record, `dateOfSeverity` will match the existing record's `dateObserved` value.